src/classes/grade.js

```diff
@@ -1,250 +1,207 @@
-        if (this.specialPage) return {}
+    setHttps (https, httpsAutoUpgrade) {
+        this.https = https
+        this.httpsAutoUpgrade = httpsAutoUpgrade
+    }
 
-        this.decisions = []
+    setPrivacyScore (score) {
+        this.privacyScore = typeof score === 'number' ? score : UNKNOWN_PRIVACY_SCORE
+    }
 
-        let beforeIndex = 1
-        let afterIndex = 1
+    addEntityBlocked (name, prevalence) {
+        if (!name) return
+        if (!prevalence) prevalence = 1
 
-        this.addDecision({change: 1, index: beforeIndex, why: 'Default grade'})
+        this.entitiesBlocked[name] = prevalence
+    }
 
-        if (this.isaMajorTrackingNetwork) {
-            beforeIndex += this.isaMajorTrackingNetwork
-            afterIndex += this.isaMajorTrackingNetwork
-            this.addDecision({
-                change: this.isaMajorTrackingNetwork,
-                index: beforeIndex,
-                why: 'Is a major tracking network'
-            })
-        } else {
-            this.addDecision({ change: 0, index: beforeIndex, why: 'Not a major tracking network' })
-        }
+    addEntityNotBlocked (name, prevalence) {
+        if (!name) return
+        if (!prevalence) prevalence = 1
 
-        // If tosdr already determined a class ranking then we map that to increase or
-        // decrease the grade accordingly. Otherwise we apply a +/- to the grade based
-        // on the cumulative total of all the points we care about. see: scripts/tosdr-topics.json
-        if (this.tosdr) {
-            if (this.tosdr.class) {
-                beforeIndex += tosdrClassMap[this.tosdr.class]
-                afterIndex += tosdrClassMap[this.tosdr.class]
-
-                this.addDecision({
-                    change: tosdrClassMap[this.tosdr.class],
-                    index: beforeIndex,
-                    why: `Has tosdr class ${this.tosdr.class}`,
-                    tosdr: this.tosdr
-                })
-            } else if (this.tosdr.score) {
-                let tosdrScore = Math.sign(this.tosdr.score)
-                beforeIndex += tosdrScore
-                afterIndex += tosdrScore
-
-                this.addDecision({
-                    change: tosdrScore,
-                    index: beforeIndex,
-                    why: `Has tosdr score ${tosdrScore}`,
-                    tosdr: this.tosdr
-                })
-            } else {
-                this.addDecision({ change: 0, index: beforeIndex, why: `No tosdr` })
-            }
-        }
+        this.entitiesNotBlocked[name] = prevalence
+    }
 
-        if (this.inMajorTrackingNetwork) {
-            beforeIndex++
-            this.addDecision({ change: 1, index: beforeIndex, why: `In major tracking network` })
-        } else {
-            this.addDecision({ change: 0, index: beforeIndex, why: `Not in major tracking network` })
-        }
+    setParentEntity (name, prevalence) {
+        this.addEntityNotBlocked(name, prevalence)
+    }
```

Comment:
**Q:** One thing I noticed in the prototype scripts was that a parent company is counted internally as a tracker not blocked: https://github.com/duckduckgo/privacy-grade/blob/9376fa575eb68f5323707160f6dc4c71d369fcdd/grade-scripts/post-process/grade.js#L102-L111

Then, each tracker that doesn't have prevalence data is set to a prevalence of 1%: https://github.com/duckduckgo/privacy-grade/blob/9376fa575eb68f5323707160f6dc4c71d369fcdd/grade-scripts/post-process/grade.js#L158

So my thinking is, it seems we're penalizing sites simply for having a parent entity, regardless of whether it's in our tracker prevalence tables or not. I'm not sure if this is intentional?